src/batched/KokkosBatched_InverseLU_Serial_Impl.hpp

```diff
@@ -0,0 +1,195 @@
+
+      const int
+        m = A.dimension(0),
+        n = A.dimension(1);
+
+      static_assert(is_vector<vector_type>::value, "value type is not vector type");      
+      static_assert(vector_type::vector_length == 4 || vector_type::vector_length == 8, 
+                    "AVX, AVX2 and AVX512 is supported");
+      const MKL_COMPACT_PACK format = vector_type::vector_length == 8 ?  MKL_COMPACT_AVX512 : MKL_COMPACT_AVX;
+
+      int r_val = 0;
+      if (A.stride_0() == 1) {
+        //mkl_dgetrfnp_compact(MKL_COL_MAJOR, 
+        //                     m, n, 
+        //                     (double*)A.data(), A.stride_1(), 
+        //                     (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+
+        //void mkl_dgetrfnp_compact (MKL_LAYOUT layout, MKL_INT m, MKL_INT n, double * ap, MKL_INT ldap, 
+        //                           MKL_INT * info, MKL_COMPACT_PACK format, MKL_INT nm);
+
+        //void mkl_dgetrinp_compact (MKL_LAYOUT layout, MKL_INT n, double * ap, MKL_INT ldap, double * work, 
+        //                           MKL_INT lwork, MKL_INT * info, MKL_COMPACT_PACK format, MKL_INT nm);
+        mkl_dgetrinp_compact (MKL_COL_MAJOR, n, 
+                              (double*)A.data(), A.stride_1(), 
+                              (double*)W.data(), n*n, 
+                              (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+
+      } else if (A.stride_1() == 1) {
+        //mkl_dgetrfnp_compact(MKL_ROW_MAJOR, 
+        //                     m, n, 
+        //                     (double*)A.data(), A.stride_0(), 
+        //                     (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+        mkl_dgetrinp_compact (MKL_ROW_MAJOR, n, 
+                              (double*)A.data(), A.stride_0(), 
+                              (double*)W.data(), n*n, 
+                              (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+      } else {
+        r_val = -1;
+      }
+      return r_val;
+    }
+#endif
+
+    template<>
+    template<typename AViewType,
+             typename WViewType>
+    KOKKOS_INLINE_FUNCTION
+    int
+    SerialInverseLU<Algo::InverseLU::Unblocked>::
+    invoke(const AViewType &A, 
+           const WViewType &W,
+           const typename MagnitudeScalarType<typename AViewType::non_const_value_type>::type tiny) {
+        static_assert(AViewType::rank == 2, "A should have two dimensions");
+        static_assert(std::is_same<typename AViewType::memory_space, typename WViewType::memory_space>::value, "A and W should be on the same memory space");
+        assert(A.span()*sizeof(typename AViewType::value_type) <= W.span()*sizeof(typename WViewType::value_type));
+
+        typedef typename AViewType::value_type ScalarType;
+
+        int B_stride_0, B_stride_1;
+        if (WViewType::rank == 1) {
+            if (std::is_same<typename WViewType::array_layout, Kokkos::LayoutRight>::value){
```

Comment:
If the work view is 1D array, layout does not matter. Why do you distinct the layout ?

Reply:
@kyungjoo-kim The work view can be a subview. For example:  ```auto ww = Kokkos::subview(w, k, Kokkos::ALL())```. 
Although ww is 1D view but it can be LayoutStride if w is LayoutLeft. 
However, if w is LayoutRight, ww is LayoutRight.
Is it reasonable?

Reply:
No it is not reasonable.